frontend/lib/src/components/elements/PlotlyChart/PlotlyChart.tsx

```diff
@@ -456,6 +466,50 @@ export function PlotlyChart({
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [plotlyFigure.layout?.dragmode])
 
+  // React to theme changes on rerun, maintaining selections, dimensions,
+  // and non-color layout properties - issue #11974
+  useEffect(() => {
+    setPlotlyFigure(prevFigure => {
+      const themed = applyTheming(initialFigureSpec, element.theme, theme)
+
+      // Merge previous non-color layout properties (margins, axis ranges, etc.)
+      themed.layout = {
+        ...prevFigure.layout,
+        ...themed.layout,
+      }
+
+      if (isSelectionActivated) {
+        // Preserve layout.selections (runtime-only field)
+        const selections = (prevFigure.layout as LayoutWithSelections)
+          .selections
+        if (selections) {
+          ;(themed.layout as LayoutWithSelections).selections = selections
+        }
+
+        // Preserve data[...].selectedpoints
+        if (prevFigure.data && themed.data) {
+          themed.data = themed.data.map((trace, idx) => {
+            const prevTrace = prevFigure.data[idx] as TraceWithSelection
+            const selPoints = prevTrace.selectedpoints
+            return selPoints !== undefined
+              ? ({ ...trace, selectedpoints: selPoints } as TraceWithSelection)
+              : trace
+          }) as unknown as typeof prevFigure.data
+        }
+      }
```

Comment:
This block feels a bit risky, need to double check with the plotly spec, but from what I remember, there might be some other parts to cover over that could otherwise lead to some surprising behaviour. However, I also don't have a good idea on how to solve this differently. Coloring is a bit broken in plotly thats why we need the replacement hack :( One potential alternative might be to only apply this to charts that don't have selections, and accept it for now that it's not updating colors for Plotly charts with selections. But no strong opinion.

Reply:
Appreciate your review! I'd agree this change feels risky - my vote would be to not include this PR in the 1.48 release and downgrade the issue from a P2 to be revisited at another point.